### PR TITLE
Gonna bring you the power! (power tools)

### PR DESCRIPTION
**Pull request by Cuboos**
:cl:
add: Added a new unique tool belt to the CE
add: Added new power tools, a hand drill that can be used as a screw driver or wrench and jaws of life which can be used as either a crowbar or wire cutters
add: Power tools can also be made from the protolathe with engineering and electromagnet 6
tweak: slightly revamped how construction/deconstruction handle sound
soundadd: added new sounds for the power tools
/:cl:

New power tools for the CE! It's like goof's golden tools but less memey and makes more sense, power tools are much faster than normal tools and take the place of two. 

Hand drill, activate to switch between between screwdriver bit and bolt bit (screwdriver/wrench)

Jaws of life, activate to switch between a cutting head and prying head, can also open powered doors like a xeno. Cutting head cuts your head off if you suicide with them. I wanted to make the prying head gib your head but i couldn't figure it out. 

Reworked how construction handles sounds and added a new "usesound" var to obj/items. This is useful if you want to add another tool or item that makes a send when it's being used. Also useful if you want to replace a sound for an item or if you have an item with a choice of sounds (like if you have two versions of a sound sound you want to play) 

CE now gets a unique white tool belt, exactly the same as a regular tool belt, just a re-skin. 

CE also spawns with the new power tools and an experimental welder, CE will be the ONLY one to spawn with these tools, might make him a more valuable target for traitors.  I figured this would be acceptable considering most heads are not antag enabled (i think) 

willing to take critique, and this might need some spell checking.

### CODE REVIEW

**Comment by PKPenguin321** [comment]
>over nine thousand lines changed

**Comment by PKPenguin321** [comment]
also CE's spawned tools shouldnt have research/material levels so R&D can't fuck him as hard

**Comment by Cuboos** [comment]
i figured there should be a way to get this later in the round, but not so easy to get just anyone can have them with a few research.

**Review by Jordie0608 — CHANGES_REQUESTED** [review]
You've messed up the line endings; you will need to convert the files back to LF

**Comment by Cuboos** [comment]
@Jordie0608 can you be more spessific? i'm not sure what i messed up

**Comment by RandomMarine** [comment]
Make them fairly expensive cargo orders instead of giving R&D MORE goodies.

**Comment by TechnoAlchemisto** [comment]
>considering most heads are not antag enabled (i think)

All heads are antag enabled except captain.

**Comment by Jazaen** [comment]
And HoS.

**Comment by optimumtact** [comment]
you have 53 files changed because your editor is switchign all line endings from unix to windows style, or potentially from windows to unix.

Unfuck your editor config and honestly probably just best to redo your changes as the easiest way to clean up the state of this PR.

**Comment by Cuboos** [comment]
Why do all my PRs keep getting fucked up :C

**Comment by nullbear** [comment]
unix, OSX, and windows all use different line endings.
i'll use this text as an example:

Windows:
```
unix, OSX, and windows all use different line endings.\r\n
i'll use this text as an example:\r\n
```

Unix or OSX:
```
unix, OSX, and windows all use different line endings.\n
i'll use this text as an example:\n
```

**Comment by Cuboos** [comment]
How come this wasn't an issue before?

**Comment by nullbear** [comment]
See this: https://help.github.com/articles/dealing-with-line-endings/

**Comment by Cuboos** [comment]
Okay, so i have the time and energy to deal with this now... so what exactly do i have to do? convert the line endings to unix? Windows? Mac?

**Comment by Cuboos** [comment]
Did that fix it?

**Review by Jordie0608 — APPROVED** [review]

File: code/game/machinery/doors/airlock.dm [hunk 1]
```diff
@@ -975,6 +975,34 @@ var/list/airlock_overlays = list()
...
+		var/time_to_open = 5
+		if(hasPower())
+			time_to_open = 50
+			playsound(src, 'sound/machines/airlock_alien_prying.ogg',100,1) //is it aliens or just the CE being a dick?
+			if(do_after(user, time_to_open,target = src))
+				open(2)
+				if(density && !open(2))
+					user << "<span class='warning'>Despite your attempts, the [src] refuses to open.</span>"
+		else
+			open(2)//hopefully this won't be redundant
```
Comment (phil235) [inline code/game/machinery/doors/airlock.dm]: copypasta and bad logic.

File: code/game/objects/items/weapons/tools.dm [hunk 2]
```diff
@@ -46,14 +47,39 @@
...
+	materials = list(MAT_METAL=150,MAT_SILVER=50,MAT_TITANIUM=25)
+	origin_tech = "materials=2;engineering=6" //done for balance reasons, making them high value for research, but harder to get
+	force = 8 //might or might not be too high, subject to change
+	throwforce = 8
+	attack_verb = list("drilled", "screwed", "jabbed")
+	toolspeed = 4
+
+/obj/item/weapon/wrench/power/attack_self(mob/user)
+	playsound(get_turf(user),"sound/items/change_drill.ogg",50,1)
+	var/obj/item/weapon/wirecutters/power/s_drill = new /obj/item/weapon/screwdriver/power
```
Comment (phil235) [inline code/game/objects/items/weapons/tools.dm]: wrong typecasting

File: code/game/objects/items/weapons/tools.dm [hunk 3]
```diff
@@ -46,14 +47,39 @@
...
+	user << "<span class='notice'>You attach the screw driver bit to the [src.name].</span>"
```
Comment (phil235) [inline code/game/objects/items/weapons/tools.dm]: screwdriver in one word

File: code/game/objects/items/weapons/tools.dm [hunk 4]
```diff
@@ -46,14 +47,39 @@
...
+	user.unEquip(src)
+	user.put_in_active_hand(s_drill)
+	qdel(src)
+
+obj/item/weapon/wrench/power/suicide_act(mob/user)
+	user.visible_message("<span class='suicide'>[user] is pressing the [src.name] against \his head, it looks like he's trying to screw \his head off")
```
Comment (phil235) [inline code/game/objects/items/weapons/tools.dm]: shouldn't this be for the screwdriver/power?

File: code/game/objects/items/weapons/tools.dm [hunk 5]
```diff
@@ -135,11 +162,40 @@
...
+
+/obj/item/weapon/screwdriver/power/suicide_act(mob/user)
+	user.visible_message("<span class='suicide'>[user] is putting the [src.name] up to \his temple, it looks like they're trying to commit suicide</span>")
+	return(BRUTELOSS)
+
+/obj/item/weapon/screwdriver/power/attack_self(mob/user)
+	playsound(get_turf(user),"sound/items/change_drill.ogg",50,1)
+	var/obj/item/weapon/wrench/power/b_drill = new /obj/item/weapon/wrench/power
+	user << "<span class='notice'>You attach the bolt driver bit to the [src.name].</span>"
+	user.unEquip(src)
```
Comment (phil235) [inline code/game/objects/items/weapons/tools.dm]: if(user.unEquip(src)) 

Just to be safe, if the item has NODROP flag somehow.

File: code/game/objects/items/weapons/tools.dm [hunk 6]
```diff
@@ -199,6 +256,32 @@
...
+	item_state = "jawsoflife"
+	origin_tech = "materials=2;engineering=6"
+	materials = list(MAT_METAL=150,MAT_SILVER=50,MAT_TITANIUM=25)
+	usesound = 'sound/items/jaws_cut.ogg'
+	toolspeed = 4
+
+/obj/item/weapon/wirecutters/power/suicide_act(mob/user)
+	user.visible_message("<span class='suicide'>[user] is the [src.name] around \his neck, it looks like \he's trying to rip \his head off!</span>")
+	playsound(loc, 'sound/items/jaws_cut.ogg', 50, 1, -1)
+	var/mob/living/carbon/C = user
```
Comment (phil235) [inline code/game/objects/items/weapons/tools.dm]: are you sure user is always a carbon?

File: code/game/objects/items/weapons/tools.dm [hunk 7]
```diff
@@ -199,6 +256,32 @@
...
+	var/obj/item/bodypart/BP = C.get_bodypart("head")
+	BP.drop_limb()// <- i have no idea why this works, but it does
```
Comment (phil235) [inline code/game/objects/items/weapons/tools.dm]: could runtime if a headless changeling uses it.

**Comment by nullbear** [comment]
yeah, the line endings are fixed.

File: code/game/objects/items/weapons/tools.dm (hunk 7 above)
Comment (Cuboos) [inline code/game/objects/items/weapons/tools.dm]: So what... should i do a if(C.get_bodypart("head")) ?

File: code/game/objects/items/weapons/tools.dm (hunk 6 above)
Comment (Cuboos) [inline code/game/objects/items/weapons/tools.dm]: who else can use a tool besides carbon?

File: code/game/objects/items/weapons/tools.dm (hunk 5 above)
Comment (Cuboos) [inline code/game/objects/items/weapons/tools.dm]: Is the even likely to happen with a tool?

File: code/game/objects/items/weapons/tools.dm (hunk 4 above)
Comment (Cuboos) [inline code/game/objects/items/weapons/tools.dm]: Yeah it probably should be... i'll fix that in a bit...

File: code/game/objects/items/weapons/tools.dm (hunk 3 above)
Comment (Cuboos) [inline code/game/objects/items/weapons/tools.dm]: Yeap... i knew there was going to be a spelling error that escaped my proofreading.

File: code/game/objects/items/weapons/tools.dm (hunk 2 above)
Comment (Cuboos) [inline code/game/objects/items/weapons/tools.dm]: oh fuck

**Comment by Cuboos** [comment]
http://i.imgur.com/fiJsjk2.png 

Why is it saying there's conflicted files? i thought we fixed this.

**Comment by nullbear** [comment]
because there are conflicting files. Someone made a PR which conflicts with your PR, and it was merged before yours was merged. This is completely unrelated to the line endings error.

**Comment by Cuboos** [comment]
If this gets merged... will it cause an issue?

**Comment by PKPenguin321** [comment]
It can't be merged if it has conflicts because it simply doesn't work that way

**Comment by Cuboos** [comment]
So what do i do?

**Comment by PKPenguin321** [comment]
Git pull origin master or something idk ask oranges on IRC

**Comment by TrustyGun** [comment]
I'm not a git wizard but by this point I would recommend that you salvage the code+icons and redo it

**Comment by Cuboos** [comment]
That's a whole hell of a lot of work to redo

**Comment by optimumtact** [comment]
There were two issues with your pr

1) you had rewritten the line endings in a bunch of files, causing a bunch of noise in the diffs, this is the issue we fixed yesterday

2) you have files that have changed on upstream master and those changes are conflicting against your own.

You need to do a merge and then fix these conflicts, open a git shell in your repo
```
git checkout PowerTools
git fetch upstream
git merge upstream/master
```
At this stage there are likely to be conflicts so use your favourite editor to resolve the conflicts in each conflicting file
```
git status will get you a list of conflicting files
<editor> path/to/conflicting/file
git add  path/to/conflicting/file
repeat for each conflict
```
finally to finish merging after you resolved and added each conflicted file
```
git commit

git push
```

File: code/game/objects/items/weapons/tools.dm (hunk 6 above)
Comment (phil235) [inline code/game/objects/items/weapons/tools.dm]: dextrous animals? maitenance drones?

File: code/game/objects/items/weapons/tools.dm (hunk 7 above)
Comment (phil235) [inline code/game/objects/items/weapons/tools.dm]: if(BP)

File: code/game/objects/items/weapons/tools.dm (hunk 6 above)
Comment (Cuboos) [inline code/game/objects/items/weapons/tools.dm]: I forgot about maintenance drones, but aren't the only dextrous animals monkeys? and aren't they also a carbon mob?

File: code/game/machinery/doors/airlock.dm (hunk 1 above)
Comment (Cuboos) [inline code/game/machinery/doors/airlock.dm]: Got any pointers on how to rewrite it?

File: code/game/objects/items/weapons/tools.dm (hunk 6 above)
Comment (phil235) [inline code/game/objects/items/weapons/tools.dm]: monkeys aren't animals, codewise. Dextrous simple_animal is a relatively new feature. The only animal using that feature now is one of the guardians.

Comment (Cuboos) [inline code/game/objects/items/weapons/tools.dm]: Soo.... what, add a conditional? if(is_type(carbon)) or something?

**Comment by Cuboos** [comment]
Well that didn't work.

**Comment by Cuboos** [comment]
Making a new PR, closing this one.

**Pull request closed without merging**